src/libraries/System.Runtime/tests/System/Text/StringBuilderTests.cs

```diff
@@ -2230,5 +2230,23 @@ public static void EqualsIgnoresMaxCapacity()
 
             Assert.True(sb1.Equals(sb2));
         }
+
+        [Fact]
+        public static unsafe void FailureOnLargeString()
+        {
+            RemoteExecutor.Invoke(() => // Uses lots of memory
+            {
+                AssertExtensions.Throws<ArgumentOutOfRangeException>("valueCount", () =>
```

Comment:
The argument name isn't ideal, but it seems not too important; this can already happen in other codepaths.